### ISSUE TITLE
[JENKINS-41044] Unable to approve a pipeline waiting on input when permissions are specified for approvers

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/blueocean-plugin#3619]
<p><b>Scope</b></p>
<ul>
	<li>Investigate what the problem is here - if you have the input up to restrict to particular users are we doing the right thing to allow this to work?
	<ul>
		<li>This needs an ATH test to ensure we keep doing the right thing.</li>
	</ul>
	</li>
	<li>Work out how to return the error to the UI (this may be a simple fix in the frontend to show the message on screen?)</li>
</ul>

<p><b>Note</b></p>
<ul>
	<li><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=vivek" class="user-hover" rel="vivek">vivek</a> we can pass this ticket on to a frontend developer to add the message to the screen if that is not already the behaviour</li>
</ul>

<p><b>Original request</b><br/>
My pipeline build has a input step that I have restricted approval from select users and I am part of those users but as I hit approve button to that input step it fails with</p>

<pre>
<span class="code-quote">"message"</span> : <span class="code-quote">"You need to be xxx, yyy, zzz to submit <span class="code-keyword">this</span>"</span>, <span class="code-quote">"code"</span> : 400, <span class="code-quote">"errors"</span> : [ ]
</pre>

<p>event though I am xxx, does blueocean not store the credentials/username of the person logged in ?</p>

<p><b>Edit 1</b> : I restrict the user that can approve the input step with the following code : </p>

<pre>
approvedInput = input id: <span class="code-quote">'dynamicInputApproval'</span>, message: <span class="code-quote">'Waiting <span class="code-keyword">for</span> EP to approve'</span>, ok: <span class="code-quote">'Approve'</span>, parameters: [
    [$class: <span class="code-quote">'ChoiceParameterDefinition'</span>, choices: <span class="code-quote">"${dynamicInput["</span>OS<span class="code-quote">"]}"</span>, description: <span class="code-quote">'Select the os your job targets'</span>, name: <span class="code-quote">'OS'</span>],
    [$class: <span class="code-quote">'BooleanParameterDefinition'</span>, defaultValue: dynamicInput[<span class="code-quote">"Docker"</span>], description: <span class="code-quote">'Enable docker workflow'</span>, name: <span class="code-quote">'Docker'</span>]
   ], submitter: <span class="code-quote">"xxx,yyy,zzz"</span>, submitterParameter: <span class="code-quote">"something"</span>
</pre>

<p>and yes, it works in classic view where it allows me (xxx) to approve it but fails on blue ocean with the above mentioned error. </p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kaulk.jpg" title="kaulk's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kaulk">kaulk</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-41044" target="_blank">Unable to approve a pipeline waiting on input when permissions are specified for approvers</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/tscherler.png" title="tscherler's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tscherler">tscherler</a>
<li><b>status</b>: Closed
<li><b>priority</b>: Major
<li><b>component(s)</b>: blueocean-plugin
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2017-02-14T11:52:59+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 4
<li><b>imported</b>: 2025-12-01
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p><b>Scope</b></p>
<ul>
	<li>Investigate what the problem is here - if you have the input up to restrict to particular users are we doing the right thing to allow this to work?
	<ul>
		<li>This needs an ATH test to ensure we keep doing the right thing.</li>
	</ul>
	</li>
	<li>Work out how to return the error to the UI (this may be a simple fix in the frontend to show the message on screen?)</li>
</ul>

<p><b>Note</b></p>
<ul>
	<li><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=vivek" class="user-hover" rel="vivek">vivek</a> we can pass this ticket on to a frontend developer to add the message to the screen if that is not already the behaviour</li>
</ul>

<p><b>Original request</b>
My pipeline build has a input step that I have restricted approval from select users and I am part of those users but as I hit approve button to that input step it fails with</p>

<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java"><span class="code-quote">"message"</span> : <span class="code-quote">"You need to be xxx, yyy, zzz to submit <span class="code-keyword">this</span>"</span>, <span class="code-quote">"code"</span> : 400, <span class="code-quote">"errors"</span> : [ ]
</pre>
</div></div>

<p>event though I am xxx, does blueocean not store the credentials/username of the person logged in ?</p>

<p><b>Edit 1</b> : I restrict the user that can approve the input step with the following code : </p>

<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">approvedInput = input id: <span class="code-quote">'dynamicInputApproval'</span>, message: <span class="code-quote">'Waiting <span class="code-keyword">for</span> EP to approve'</span>, ok: <span class="code-quote">'Approve'</span>, parameters: [
                    [$class: <span class="code-quote">'ChoiceParameterDefinition'</span>, choices: <span class="code-quote">"${dynamicInput["</span>OS<span class="code-quote">"]}"</span>, description: <span class="code-quote">'Select the os your job targets'</span>, name: <span class="code-quote">'OS'</span>],
                    [$class: <span class="code-quote">'BooleanParameterDefinition'</span>, defaultValue: dynamicInput[<span class="code-quote">"Docker"</span>], description: <span class="code-quote">'Enable docker workflow'</span>, name: <span class="code-quote">'Docker'</span>]
                           ], submitter: <span class="code-quote">"xxx,yyy,zzz"</span>, submitterParameter: <span class="code-quote">"something"</span>
</pre>
</div></div>

<p>and yes, it works in classic view where it allows me (xxx) to approve it but fails on blue ocean with the above mentioned error. </p></pre>
</details>
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/michaelneale.png" title="michaelneale's avatar" /> michaelneale:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-41044?focusedId=283202&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-283202">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kaulk" class="user-hover" rel="kaulk">kaulk</a> can you provide a sample of your pipeline input step where you have restricted this? </p>

<p>Also, I assume you can approve it from the regular classic Jenkins screen? </p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kaulk" class="user-hover" rel="kaulk">kaulk</a> can you provide a sample of your pipeline input step where you have restricted this? </p>

<p>Also, I assume you can approve it from the regular classic Jenkins screen? </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/michaelneale.png" title="michaelneale's avatar" /> michaelneale:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-41044?focusedId=283204&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-283204">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=vivek" class="user-hover" rel="vivek">vivek</a> I suspect this may be api related, somehow the users context isn't being used, but it is something we could and should probably have a test for (it may be something else... just keep eyes open). </p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=vivek" class="user-hover" rel="vivek">vivek</a> I suspect this may be api related, somehow the users context isn't being used, but it is something we could and should probably have a test for (it may be something else... just keep eyes open). </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kaulk.jpg" title="kaulk's avatar" /> kaulk:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-41044?focusedId=283220&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-283220">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=michaelneale" class="user-hover" rel="michaelneale">michaelneale</a> - please see edit 1.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=michaelneale" class="user-hover" rel="michaelneale">michaelneale</a> - please see edit 1.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/michaelneale.png" title="michaelneale's avatar" /> michaelneale:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-41044?focusedId=283223&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-283223">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kaulk" class="user-hover" rel="kaulk">kaulk</a> thanks! should be easy enough to fix. </p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kaulk" class="user-hover" rel="kaulk">kaulk</a> thanks! should be easy enough to fix. </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/vivek.png" title="vivek's avatar" /> vivek:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-41044?focusedId=287699&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-287699">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Probably duplicate of <a href="https://issues.jenkins-ci.org/browse/JENKINS-41421" class="external-link" target="_blank" rel="nofollow noopener">https://issues.jenkins-ci.org/browse/JENKINS-41421</a> and PR <a href="https://github.com/jenkinsci/blueocean-plugin/pull/820" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/blueocean-plugin/pull/820</a>.</p>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tscherler" class="user-hover" rel="tscherler">tscherler</a> over to you<img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>
</pre>
</details>

</ul>
</details>

<p>Probably duplicate of <a href="https://issues.jenkins-ci.org/browse/JENKINS-41421" class="external-link" target="_blank" rel="nofollow noopener">https://issues.jenkins-ci.org/browse/JENKINS-41421</a> and PR <a href="https://github.com/jenkinsci/blueocean-plugin/pull/820" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/blueocean-plugin/pull/820</a>.</p>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tscherler" class="user-hover" rel="tscherler">tscherler</a> over to you<img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `is_duplicated_by` from Jira: <a href="https://github.com/jenkinsci/blueocean-plugin/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-41421%22">JENKINS-41421</a>]</i>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `epic` from Jira: <a href="https://github.com/jenkinsci/blueocean-plugin/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-38490%22">JENKINS-38490</a>]</i>
